Add recipe for evil-neo

p-conrad opened the pull request:
### Brief summary of what the package does

Evil-Neo allows you to use Evil's keybindings with the German [Neo keyboard layout](https://www.neo-layout.org) by remapping the hjkl keys to Neo's snrt keys. The keys are mapped in their own minor mode, enabling the user to turn it on and off as needed. It aims to preserve Evil's distinction between motion, normal, and visual states, mapping keys only in the modes they were originally defined in, where possible. The keybindings are exhaustive, with all keybindings defined in `evil-motion-state-map`, `evil-normal-state-map`, and `evil-visual-state-map` taken into account at the time of authoring this package.

A complete overview of the mapped keys is provided in the README.

### Direct link to the package repository

https://git.sr.ht/~p-conrad/evil-neo

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`evil-neo.el` with package-lint 20240507.1722:
```
2 issues found:
41:0: error: Customization groups should not end in "-mode" unless that name would conflict with their parent group.
97:17: error: This key sequence is reserved (see Key Binding Conventions in the Emacs Lisp manual)
```

p-conrad commented:
Thank you for your review and apologies for the issues. The first issue was introduced when I fixed the compilation warnings and forgot to run package-lint again afterward. I have addressed it and pushed my fixes to the repository.

Regarding the second issue, from my understanding, it concerns the use of the reserved keybinding` C-h`. However, this occurs only in the chained binding `C-w C-h`, which Evil also defines. Therefore, I consider it a false positive. I originally intended to mention this in my pull request, but it slipped my mind.

Again, thank you for your time and feedback.

riscy commented:
Thanks for clarifying - confirmed evil does bind this, and I think evil packages are in a category where we make certain small concessions around keybinding conventions to accommodate vi(m) muscle memory.

Related (but not overlapping) packages in this space -

- https://melpa.org/#/evil-dvorak
- https://melpa.org/#/evil-colemak-basics
- https://melpa.org/#/evil-colemak-minimal

There may be some information sharing or standards that might be shared across these different packages, but this looks good to merge.  And, welcome to MELPA :)
